ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
In L81, the adjusted index of a column is the same as original index when no pruning happened. Will index be changed in other places?